src/components/shapes/attributes.js

```diff
@@ -71,13 +75,41 @@ module.exports = {
             'the date to unix time in milliseconds.'
         ].join(' ')
     }),
+    xsizemode: {
+        valType: 'enumerated',
+        values: ['data', 'pixel'],
+        dflt: 'data',
```

Comment:
Ah, I hadn't thought of this before, but `'data'` mode applies not just to axis scaling but also plot fraction scaling, right? ie when `xref='paper'` this option is still relevant, right? So how about we change `'data'` to `'scaled'`? That seems like it encompasses both cases and differentiates them from `'pixel'` sizing...

Reply:
Good point. I'll change that.